subprojects/docs/src/docs/userguide/buildEnvironment.adoc

```diff
@@ -13,135 +13,216 @@
+Specify password for authenticating to a remote Gradle build cache.
+`gradle.wrapperUser=(myuser)`::
+Specify user name to download Gradle distributions from servers using HTTP Basic Authentication. Learn more in <<sec:authenticated_download>>.
+`gradle.wrapperPassword=(mypassword)`::
+Specify password for downloading a Gradle distribution using the Gradle wrapper.
+`gradle.user.home=(path to directory)`::
+Specify the Gradle user home directory.
 
-You can also set system properties in the `gradle.properties` file. If a property name in such a file has the prefix “`systemProp.`”, like “`systemProp.propName`”, then the property and its value will be set as a system property, without the prefix. In a multi project build, “`systemProp.`” properties set in any project except the root will be ignored. That is, only the root project's `gradle.properties` file will be checked for properties that begin with the “`systemProp.`” prefix.
+In a multi project build, “`systemProp.`” properties set in any project except the root will be ignored. That is, only the root project's `gradle.properties` file will be checked for properties that begin with the “`systemProp.`” prefix.
 
-++++
-<sample id="properties" dir="userguide/tutorial/properties" title="Setting properties with a gradle.properties file">
-            <sourcefile file="gradle.properties"/>
-            <sourcefile file="build.gradle"/>
-            <output args="-q -PcommandLineProjectProp=commandLineProjectPropValue -Dorg.gradle.project.systemProjectProp=systemPropertyValue printProps"/>
-        </sample>
-++++
+[[sec:gradle_environment_variables]]
+=== Environment variables
 
+The following environment variables are available for the `gradle` command. Note that command-line options and system properties take precedence over environment variables.
 
-[[sub:checking_for_project_properties]]
-==== Checking for project properties
+`GRADLE_OPTS`::
+Specifies <<command_line_interface, command-line arguments>> to use when starting the Gradle client. This can be useful for setting the properties to use when running Gradle.
+`GRADLE_USER_HOME`::
+Specifies the Gradle user home directory (which defaults to `$USER_HOME/.gradle` if not set).
+`JAVA_HOME`::
+Specifies the JDK installation directory to use.
 
-You can access a project property in your build script simply by using its name as you would use a variable. If this property does not exist, an exception will be thrown and the build will fail. If your build script relies on optional properties the user might set, perhaps in a `gradle.properties` file, you need to check for existence before you access them. You can do this by using the method `hasProperty('propertyName')` which returns `true` or `false`.
+[[sec:project_properties]]
+=== Project properties
 
-[[sec:accessing_the_web_via_a_proxy]]
-=== Accessing the web via a proxy
+You can add properties directly to your api:org.gradle.api.Project[] object via the `-P` command line option.
```

Comment:
Potentially ask give an example on how to use it in the build script or reference the relevant methods.

Reply:
It's explained below by example. done